src/datatypes/datetime.ts

```diff
@@ -75,33 +75,470 @@ function truncateLuxonDateTime(luxonDT: LuxonDateTime, unit: DateTimeUnit) {
   return luxonDT.startOf(unit);
 }
 
-export class DateTime {
-  year: number | null;
-  month: number | null;
-  day: number | null;
+/*
+ * Base class for Date and DateTime to extend from
+ * Implements shared functions by both classes
```

Comment:
(After reviewing...)  Yeah, this is definitely less wonky. I like it!

Reply:
Less wonky, agreed. Still slightly wonky nonetheless, but that's a problem for future us :)